processor/deltatocumulativeprocessor/internal/streams/limit.go

```diff
@@ -23,21 +28,27 @@ type LimitMap[T any] struct {
 }
 
 func (m LimitMap[T]) Store(id identity.Stream, v T) error {
-	_, ok := m.Map.Load(id)
-	avail := m.Map.Len() < m.Max
-	if ok || avail {
-		return m.Map.Store(id, v)
-	}
+	_, exist := m.Map.Load(id)
 
-	errl := ErrLimit(m.Max)
-	if m.Evictor != nil {
-		gone := m.Evictor.Evict()
-		if err := m.Map.Store(id, v); err != nil {
-			return err
+	var errEv error
+	// if not already tracked and no space: try to evict
+	if !exist && m.Map.Len() >= m.Max {
+		errl := ErrLimit(m.Max)
+		gone, ok := m.Evictor.Evict()
```

Comment:
for a future PR: it would be nice to have an eviction counter, if there isn't one already